Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -40,17 +114,17 @@ function CreateProjectGuidAndPathMap
 function IsComponentFormatValid($tokens)
 {
     #expected format for component '[SonarQube project key]:[SonarQube project value]:[MSBuild project guid]:[file name relative to MSBuild project file path]'
-    $isFormatValid = $false
 
     if (!$tokens)
     {
         Write-Verbose "IsComponentFormatValid: tokens is invalid"
-        return $isFormatValid
+        return $false
```

Comment:
I refactored this to GetComponetGuid that parses and extracts the guid. I decided to throw exceptions instead of emitting warnings because the warnings are not user actionable. Fail fast...